src/cpp/core/system/PosixSystem.cpp

```diff
@@ -2403,8 +2403,28 @@ Error userBelongsToGroup(const User& user,
                          bool* pBelongs)
 {
    *pBelongs = false; // default to not found
+
+   // first we try to use getgrouplist(3) to determine if the user is a
+   // member of the target groupName
+   std::vector<group::Group> groups;
+   Error error = group::userGroups(user.getUsername(), &groups);
+   if (error)
+      return error;
+
+   // scan the user's groups to see if they are a member of the target group
+   *pBelongs = std::find_if(groups.begin(),
+                            groups.end(),
+                            [&](const group::Group& group)
+                            { return group.name == groupName; }) != groups.end();
+
+   // if the user belongs, then we can return early
+   // otherwise, we fallback to the legacy implementation
+   if (pBelongs)
```

Comment:
```suggestion
   if (*pBelongs)
```

Presumedly we want to check the pointer value rather than the pointer itself?